presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -201,7 +202,8 @@ protected RelationPlan visitTable(Table node, Void context)
             }
 
             List<Symbol> outputSymbols = outputSymbolsBuilder.build();
-            PlanNode root = TableScanNode.newInstance(idAllocator.getNextId(), handle, outputSymbols, columns.build());
+            boolean isDeleteTarget = analysis.isDeleteTarget(createQualifiedObjectName(session, node, node.getName()));
+            PlanNode root = TableScanNode.newInstance(idAllocator.getNextId(), handle, outputSymbols, columns.build(), isDeleteTarget);
```

Comment:
is there any way to test it?

Reply:
I didn't find a way. DELETE queries are rejected in `TestLogicalPlanner`